### DISCUSSION TITLE
starlight app deployed at cloudflare pages failed on pagefind load

### DISCUSSION THREAD

**Discussion by fujohnwang** [withastro/starlight#2267] [Q&A]
I don't know whether it's a cloudflare pages issue or starlight issue, maybe it's not a bug of starlight, but I don't know which place is the proper one to tell.

An app built with starlight failed to load pagefind.js dynamically:

<img width="1911" alt="image" src="https://github.com/user-attachments/assets/d1afa499-05cf-4194-946e-cb4835e3e5f2">

But as I saw thru local build folder, the pagefind.js is there in dist/ folder.

**Comment by delucis** [comment]
One thing I might try is to check if you have Cloudflare’s “Rocket Loader” feature enabled. We’ve seen that break JavaScript loading before although I think the error looked a bit different.

Do you have a version of deployed site we can look at to debug?

**Reply by fujohnwang** [reply]
I have tested in another way, it's a sure thing that's the issue on cloudflare side, sorry to bother.

Maybe I will build entirely on local machine instead of depending on cf's CICD 🤣
